api/v1beta1/machine_types.go

```diff
@@ -59,6 +59,22 @@ const (
 	// This annotation can be set on BootstrapConfig or Machine objects. The value set on the Machine object takes precedence.
 	// This annotation can only be used on Control Plane Machines.
 	MachineCertificatesExpiryDateAnnotation = "machine.cluster.x-k8s.io/certificates-expiry"
+
+	// NodeRoleLabelDomain is one of the CAPI managed Node label domains.
+	NodeRoleLabelDomain = "node-role.kubernetes.io/"
```

Comment:
Based on the proposal we want to treast this one as a prefix and the other two as domains
(of course also affects the controller implementation)